managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java

```diff
@@ -3393,4 +3393,26 @@ public void testCancellationOfScheduledTasks() throws Exception {
         assertTrue(timeoutTask2.isCancelled());
         assertTrue(checkLedgerRollTask2.isCancelled());
     }
+
+    @Test
+    public void testInactiveLedgerRollOver() throws Exception {
+        int inactiveLedgerRollOverTimeMs = 5;
+        ManagedLedgerFactoryConfig factoryConf = new ManagedLedgerFactoryConfig();
+        ManagedLedgerFactory factory = new ManagedLedgerFactoryImpl(metadataStore, bkc);
+        ManagedLedgerConfig config = new ManagedLedgerConfig();
+        config.setInactiveLedgerRollOverTime(inactiveLedgerRollOverTimeMs, TimeUnit.MILLISECONDS);
+        ManagedLedgerImpl ledger = (ManagedLedgerImpl) factory.open("rollover_inactive", config);
+        ManagedCursor cursor = ledger.openCursor("c1");
+
+        int totalAddEntries = 5;
+        for (int i = 0; i < totalAddEntries; i++) {
+            String content = "entry"; // 5 bytes
+            ledger.addEntry(content.getBytes());
```

Comment:
it would be interesting to see what happens in a failure scenario, when the addEntry is rejected. 
From your implementation we consider the topic active even if a write fails. it makes sense to me. just asking to add a unit test for that

Reply:
I am not sure that this is an important aspect, worth to add a test case.

why is it so important to you ?
if a write fails the ledger will be rolled over in any case

Reply:
it marks activity if it receives a write without depending on failures or introducing any complexity about the failures. so,  I don't see any specific need for that test.